### DISCUSSION TITLE
Q-Input and  v-model.trim and clearable  does not work on Q2

### DISCUSSION THREAD

**Discussion by zanzara** [quasarframework/quasar#10362] [General - Components / Directives / etc]
IMO it seems to be a bug:
If i use this:
```vue
     <q-input
         v-model.trim="v$.bildUrl.$model" 
        :error="v$.bildUrl.$error"
        :error-message="getPicErrMsg(v$.bildUrl)"
         dense
         lazy-rules
         clearable
         autofocus
         :hint="$t('gwsCol.Bild.hint')"
   />
```
it throws me this error when I click on `clearable`:
![Bildschirmfoto zu 2021-08-12 15-18-12](https://user-images.githubusercontent.com/7381054/129203727-be2be1ea-80e5-4475-84ec-cade5632db40.png)

If I  use `v-model` w/o `.trim` it works.
Can anyone confirm? Or where is my mistake?  (it works in Q1)

(Info:   » Pkg quasar........ v2.0.3    » Pkg @quasar/app... v3.0.4 )

**Comment by ontwikkelfabriek** [comment]
@zanzara 

Please post the `script` part of the code. Even better create a codepen.io to demo your issue for better support.

**Reply by zanzara** [reply]
@ontwikkelfabriek Very simple. Pls. go to [https://quasar.dev/vue-components/input#clearable](url) , open the codepen icon there and exchange the first v-model of the first q-input with `v-model.trim` and you'll get the error:

![Bildschirmfoto zu 2021-08-12 18-29-59](https://user-images.githubusercontent.com/7381054/129233699-5d82b240-2edb-4514-a3da-ab60f2ed013d.png)

So it's a bug, not a feature. ;-)

**Reply by ontwikkelfabriek** [reply]
@zanzara 

Eum. I just did what you said and I don't get an error. It works...trim and all.
https://codepen.io/ontwikkelfabriek/pen/qBmvqEv

btw the link your posted is not correct when you click on it.

**Reply by zanzara** [reply]
@ontwikkelfabriek   Sorry for worng url.
But with your example it doesn't work aswell.
![Bildschirmfoto vom 2021-08-12 20-35-49](https://user-images.githubusercontent.com/7381054/129250802-287d2dec-3459-4117-b39e-d5a8664ac158.png)

did you click the clearable cross? (Im using chrome browser)

**Reply by ontwikkelfabriek** [reply]
@zanzara 

I think you should try another browser(try open chrome with privacy mode, without plugins, because you have plugins that mess with JS on sites)/system/computer/location because it **really** works...without error

Incognito/privacy mode chrome:

- Windows, Linux, or Chrome OS: Press Ctrl + Shift + n.
- Mac: Press ⌘ + Shift + n.

**Reply by zanzara** [reply]
It's getting somehow weird.
This is on latest firefox:
![Bildschirmfoto vom 2021-08-12 21-26-31](https://user-images.githubusercontent.com/7381054/129256961-23cc70c1-8ff4-4862-96eb-c81fe3ca17c2.png)

Does also not work

**Reply by zanzara** [reply]
@ontwikkelfabriek  and this is on privacy mode chrome. the same error:
![Bildschirmfoto vom 2021-08-12 21-29-21](https://user-images.githubusercontent.com/7381054/129257386-7fd6e229-4148-47c4-8171-66b1a0b4d7af.png)

**Reply by zanzara** [reply]
@ontwikkelfabriek  and i tried it on a vanilla chromium. The same error.
IMO this got nothing to do with private mode or other plugins. To me it's crystal clear an error so far.

**Reply by zanzara** [reply]
@ontwikkelfabriek  So let's go back to start. One last question:
Can you confirm , that you hit the `x-cross-clear` button on the right side of the input-field in your test?
(Only then I get the error)

**Reply by ontwikkelfabriek** [reply]
@ontwikkelfabriek 

Awww sorry. No I never clicked on that button. When I do, I get the same error.

 Clicking the cross must make the **model** value `null` or `undefined` instead of `empty string`....**bug** in **clearable** property?  

Can think of a way to work around the bug yourself? ( use a append right slot , create your own clear button with icon and with event handler that will set the v-model to `''` ( empty string).

**Reply by zanzara** [reply]
@ontwikkelfabriek  The question is, should `clearable` property really make the model value `null`?
Shouldn't it be more like setting it to `''` emtpy string?

I'm convinced this  is a bug in `clearable` property.
So I would transfer this discussion post here to to the issue tab (also beacuse in Q v1 this works w/o raising an error).

For the moment I refactored al my `v-model.trim` s to `v-model` only, where `q-input` has this prop. Maybe there could be soon a fix, then I'll reject my changes.

**Reply by ontwikkelfabriek** [reply]
@zanzara 

> I'm convinced this is a bug in clearable property.

Yes me too. 

You could be very helpful to Quasar community by creating a Github issue for this...

**Comment by qxygene** [comment]
any update on this?

**Reply by pdanpdan** [reply]
@qxygene interesting, your very helpful multiple contribution missed the only place where there was an update
https://github.com/quasarframework/quasar/issues/10376#issuecomment-1108320740
